test/TesterInternal/StorageTests/PersistenceProviderTests.cs

```diff
@@ -131,43 +131,101 @@ public void Dispose()
+                nameof(PersistenceProvider_Azure_ChangeReadFormat),
+                nameof(stringLength), stringLength == null ? "default" : stringLength.ToString(),
+                nameof(useJsonForWrite), useJsonForWrite,
+                nameof(useJsonForRead), useJsonForRead);
 
-            IStorageProvider store = new AzureTableStorage();
-            providerCfgProps.Add("DataConnectionString", StorageTestConstants.DataConnectionString);
-            providerCfgProps.Add("UseJsonFormat", "true");
-            var cfg = new ProviderConfiguration(providerCfgProps, null);
-            await store.Init(testName, storageProviderManager, cfg);
+            var grainState = TestStoreGrainState.NewRandomState(stringLength);
+            var grainId = GrainId.NewId();
 
-            await Test_PersistenceProvider_WriteRead(testName, store);
+            var store = await InitAzureTableStorageProvider(useJsonForWrite, testName);
+
+            grainState = await Test_PersistenceProvider_WriteRead(testName, store,
+                grainState, grainId);
+
+            store = await InitAzureTableStorageProvider(useJsonForRead, testName);
+
+            await Test_PersistenceProvider_Read(testName, store, grainState, grainId);
         }
 
-        [Fact, TestCategory("Functional"), TestCategory("Persistence"), TestCategory("Azure")]
-        public void AzureTableStorage_ConvertToFromStorageFormat()
+        [Theory, TestCategory("Functional"), TestCategory("Persistence"), TestCategory("Azure")]
+        [InlineData(null, "true", "false")]
+        [InlineData(null, "false", "true")]
+        [InlineData(15 * 32 * 1024 - 256, "true", "false")]
+        [InlineData(15 * 32 * 1024 - 256, "false", "true")]
+        public async Task PersistenceProvider_Azure_ChangeWriteFormat(int? stringLength, string useJsonForFirstWrite,
+            string useJsonForSecondWrite)
         {
-            TestStoreGrainState initialState = new TestStoreGrainState { A = "1", B = 2, C = 3 };
-            AzureTableStorage.GrainStateEntity entity = new AzureTableStorage.GrainStateEntity();
-            var storage = new AzureTableStorage();
+            var testName = string.Format("{0}({1} = {2}, {3} = {4}, {5} = {6})",
+                nameof(PersistenceProvider_Azure_ChangeWriteFormat),
+                nameof(stringLength), stringLength == null ? "default" : stringLength.ToString(),
+                nameof(useJsonForFirstWrite), useJsonForFirstWrite,
+                nameof(useJsonForSecondWrite), useJsonForSecondWrite);
+
+            var grainState = TestStoreGrainState.NewRandomState(stringLength);
+            var grainId = GrainId.NewId();
+
+            var store = await InitAzureTableStorageProvider(useJsonForFirstWrite, testName);
+
+            var state = await Test_PersistenceProvider_WriteRead(testName, store,
+                grainState, grainId);
+
+            store = await InitAzureTableStorageProvider(useJsonForSecondWrite, testName);
```

Comment:
I think we need to use a new state for the second write.  Imagine that there is a bug and the state is being written to the second format properties but are not cleared from the first format properties.  This test would pass only because we're writing the same data to both sets of properties.  We want to make sure that not only is data being written to the new formats properties, but that it's also being cleared from the old format's properties.